### DISCUSSION TITLE
The English translation for Dominican Office (and Mass) for 1962?

### DISCUSSION THREAD

**Discussion by aklusman** [DivinumOfficium/divinum-officium#3598] [General]
I was wondering what English translation was to be used for the Dominican Office (and Mass).  

Was the idea to take the English from 1967?  Is there any worry or concern with regards to the copyright?  It looks like the Psalms, Canticle of Habakuk, and some excerpts from the Psalms use the Grail translation, scriptural passages from not Tobit/Judith are from the RSV.  

I just wanted to be clear on this before I go too far in resolving some of the OP office's issues.  Thank you!

**Comment by jason-gislason** [comment]
Aklusman,

Did you ever get a response to your question on the English translation of the Divine Office?  I have a copy of the 1967 OP office from Ireland.  I was thinking I'd like to make the same changes you are suggesting.  Some of the "English" translation is the Latin.  Let me know how you are progressing on the changes.  I'd be happy to contribute.

Best,
Jason

**Reply by aklusman** [reply]
Hey Jason,
I've not heard any advice on my question.  To avoid any copyright issues for the site, I figured I would attempt to utilize the public domain translations as far as possible.  The only solution to getting the 'unique' OP texts would be commissioning a translation, where necessary.  That, however, given my current work on things, is probably a few years down the road, haha.

I have begun (on my local machine - I have yet to upload anything to the site) working on getting the Latin 1962 texts transcribed into a usable format.  Breviarium SOP blog posted the texts for 1909 (Latin), 1962 (Latin), and 1967 (English), however, the OCR is not that great, so it's required me going through and cleaning up the texts to get them usable to upload here.  My hold up on getting those cleaned up texts posted to here (or even to my branch of the project - which I think is how you're "supposed" to use GitHub, haha) is I am also trying to figure out the exact workings of the code to draw these points out - which is a task in and of its own.

That said, the state of the Dominican Office on DO is that it is simply just a copy of the Roman.  It has not really been changed.  Compline works for the season outside of Lent and Eastertide (again, still working on figuring out how the code works and draws from different sources).

I am happy to receive help.  I think the biggest thing at this point is transcribing the propers for upload.  I can focus on figuring out how the site can use the propers (and the easiest way to integrate the OP into the structure) and then post an update here?  Fun stuff!

**Reply by jason-gislason** [reply]
It looks like we were working on the same effort.  We should divide and conquer.  Here are a couple of thoughts:

- Since we have the 1967 English version, we can assume that translation is correct.  From praying it, there appears to be minor errors but most are corrected in the margins.  We can just use the images for that version to OCR.  
- As indicated by mbab below, we should be good on all translations for common texts.  Using Douay-Rheims ensures that there are no copyright issues.  
- For translations of the OP specific text, we can either use the 1967 version or translate it ourselves.  My Latin is very rusty.  It has been almost 40 years since I took it in school.  I've been refreshing it but I don't trust my accuracy!  I have been able to get Chat GPT to translate the Latin text in the style of Douay-Rheims with good success.  With respect to copyrights, it looks like Browne and Nolan Limited went out of business a number of years ago.  Therefore, the text may be in the public domain.  According to US law (this was published in Ireland), anything published prior to 1978 gets a 28 year protection.  The copyright owner can extend for 60+ years if they apply for the extension.  However, all that may not be relevant.  
- With respect to OCR of the Latin text, I found i2ocr (https://www.i2ocr.com/) seemed to give the best OCR results for the platforms I tested.  Adobe and others struggled due to the lack of Latin as a source language.  I even tried Chat GPT 4.  Unfortunately, they haven't implemented a new enough version of Tesseract to include Latin.  
- With respect to the readings for psalms, gospels, homilies and other texts, we should verify the same versus are used on the same days.  I noticed slight difference between the 1909 (Cormier) breviary image and the Tridentine 1906 version on DO.  I wasn't sure if that carried over to the 1962/67 version.
- With respect to figuring our the code, have you considered copying the code into Chat GPT and asking it to comment the code?  We've done that at work with some older code with a lot of success.  However, I haven't tried it here.  I agree that the code is difficult to understand at times.  Great work but not terribly well commented!

On a more philosophical note, my intention is to translate the Cormier (1909) version to English.  The images are poor.  Therefore, the OCR is painful.  What is your intention for translating/providing the OP version?  We have access to a 1967 English translation.  Therefore, getting it on DO is instructive but doesn't add a lot of utility to my pray life.  The Cormier version apparently is almost identical to Humbert's 1200's version.  Having that version translated would enrich my prayer life a lot.

**Reply by aklusman** [reply]
Hi Jason,
I'm definitely down to divine and conquer.  

Agreed on bullet point 1.  
Re: bullet point 2 - Yes, we should be good on that front, as well.  I had noticed when the OP Compline was initially deployed, it used the translation from 1967OP.  I think that is resolved, although I'd have to double check.
Point 3:  I think (and I'm no lawyer) we'd probably be safe to either use the 1967 version, or use the 1967 to inform our translations.  I'm not sure if that's "legally cheating", though, haha.
Re: point 4 - I've been using Rescribe (https://latinocr.org/).  It has worked OK, but has definitely required some touching up.  Maybe 75% accurate, I'd say?  So, decent, but not perfect.
Re: readings of psalms/gospels/homilies - they are not exactly the same as the Roman office.  I have noticed that the 1967OP Matins sermons are typically shorter than the Roman, but drawn from the same texts, sometimes.  As well, a number of the lessons are different between Roman and 1967OP.  It's a little frustrating.  
With the figuring out the code - I haven't done what you've mentioned, although it sounds like a good idea.  To be honest, once I realized the effort in getting the 1967OP particular texts typed in, I focused a lot more on that than I did on figuring out the code (since I'd need the accurate texts in Latin and English to be able to mess with the code in useful ways).  

As to your philosophical note:  Getting the 1909 Cormier version to English and running on Divinum Officium is my goal as well, to be honest.  Currently, I use the 1967OP to pray Lauds, Vespers, and Compline daily, although I've only really been doing it reliably the last few months.  Slowly but surely I've been figuring out the nuances and how-to of praying 1967OP.  Ideally, one day, I will be able to pray the 1909 Cormier.  But, I figured that there's probably a "market" for 1967OP (predominantly, I assume, among traditional Lay Dominicans).  I also think (and hope) that by getting a handle of the "easier" 1967OP, it can set the stage for implementing the 1909 Cormier as another option.  I'm guessing the texts are pretty much the same, it just gets a lot more complex with the older (better) calendar and rubrics.  But, as you mention, we have the 1967OP, so I've been using that as my springboard to get more familiar with the Dominican Office, and to eventually 'graduate' to the (better) Cormier.  I am in the same boat re: having the Cormier on here and translated would be a boon to the prayer life.

All that said, I will continue the transcription work on 1967OP and try to get it up to GitHub in a usable format (I've got decent progress on the Sanctoral side of thigns).  Once that happens, I'll try to graduate to the code...  I am, however, happy to take advice/instruction if there's a better way going forward!

**Comment by mbab** [comment]
Just 2 remarks.
1. Even Latin version of Dominican Office is very early stage.
2. All versions use same English translation of common text as psalms, readings, etc.

**Reply by jason-gislason** [reply]
Thank you.  I would definitely want to use the same translations for the common text as you described.  Do you know if there is any documentation for the code?  I've been able to figure out some of it.  However, the some of the text concatenation and "global" variables are hard to trace.

**Reply by mbab** [reply]
The coded is self documented ;-)
In true I'm working slowly on it, trying at least make it clean, terse, modularized, more perlish and avoid repetitions and use of globals (at least use `our` declaration) and whole project more data driven, but unfortunately I'm not best with adding comments.
How data is taken is described  https://www.divinumofficium.com/www/horas/Help/technical.html but some parts of this can be outdated.

**Reply by jason-gislason** [reply]
I totally understand.  Thank you for working on the code.  Refactoring someone else's code can be much harder and more labor intensive than writing the original.  Perl is especially challenging since there are so many options for doing the same thing.  I'll take a look at the link you posted to learn more.  Thanks again for doing the work you are doing.  The page is a blessing!

**Reply by mbab** [reply]
> 1. Even Latin version of Dominican Office is very early stage.

Started from Passion week I'm adding Latin tempora data for OP officium (in TemporaOP folder)  .
One can start filling English (and other vernaculars) translations.
